### ISSUE TITLE
Let sysadmin populate db with predefined list of crops

### ISSUE THREAD

**Issue by alexgleason** [HarvestHub/GardenHub#111]
The master list of crops is stored in the database but will likely not be very different between installs. So, it would be useful to provide a way to populate this list initially.

I'm thinking of offering a Django management command like `python manage.py populate_starter_crops` which would create all the crops. My only concern is that running it twice would create duplicate crops and it would be very hard to undo that action.

Another option is to create these crops within the first migration file. That way they'll only be created once per fresh install. The downside is that if someone doesn't want *any* of these crops they'll have to delete them all.

**Comment by alexgleason** [comment]
@marykatefain can you send me the list of crops you made?

**Comment by marykatefain** [comment]
It uploading to Uptwinkles drive -> CW -> Gardenhub -> Crops

**Comment by alexgleason** [comment]
this is the list you provided:

```
crops_list = [
    'acorn squash',
    'alfalfa spouts',
    'amaranth',
    'apricots',
    'artichoke',
    'asparagus',
    'avocado',
    'bananas',
    'barley',
    'basil',
    'bay leaf',
    'bean sprouts',
    'beets',
    'blackberries',
    'blueberries',
    'bok choy',
    'broccoli',
    'brussels sprouts',
    'butternut squash',
    'cabbage',
    'caper',
    'carrots',
    'cauliflower',
    'cayenne peppers',
    'celery',
    'chamomile',
    'cherries',
    'cherry tomatoes',
    'chestnuts',
    'chickpeas',
    'chives',
    'cilantro',
    'collard greens',
    'coriander',
    'corn',
    'cranberries',
    'cucumber',
    'daisy',
    'daylily',
    'dill',
    'eggplant',
    'elderberries',
    'fennel',
    'garlic',
    'ginger',
    'gourds',
    'grapefruit',
    'green apples',
    'green beans',
    'green grapes',
    'green pepper',
    'habanero pepper',
    'honeydew',
    'hops',
    'hot peppers',
    'huckleberries',
    'jalapeno peppers',
    'kale',
    'kiwi',
    'kohlrabi',
    'leeks',
    'lentils',
    'lettuce',
    'luffa',
    'marigold',
    'mustard',
    'okra',
    'olives',
    'onion',
    'oranges',
    'oregano',
    'oyster mushrooms',
    'parsley',
    'peach',
    'peanuts',
    'pears',
    'peas',
    'peppermint',
    'pineapple',
    'portobello mushrooms',
    'potatoes',
    'pumpkin',
    'purple grapes',
    'quinoa',
    'radish',
    'raspberries',
    'red apple',
    'red lettuce',
    'red onions',
    'red pepper',
    'rosemary',
    'sage',
    'scallions',
    'sesame',
    'shallots',
    'shitake mushrooms',
    'snap peas',
    'snow peas',
    'soy beans',
    'spinach',
    'starfruit',
    'strawberries',
    'sunflower',
    'sweet corn',
    'sweet potatoes',
    'thai basil',
    'thyme',
    'tomatillo',
    'tomatoes',
    'truffle',
    'tulips',
    'walnuts',
    'watermelon',
    'white onion',
    'yams',
    'yellow onion',
    'yellow peppers',
    'zucchini'
]
```

some of the names are singular and some are plural. which should they be?

**Comment by marykatefain** [comment]
Yeah so I knew this would be an issue. Here is what I ended up doing: Whatever sounds better.

A lot of crops sound better singular (Ex. "Please harvest my basil and cauliflower"), but some sound REALLY WEIRD. ("Go harvest my strawberry and snap pea" for example). I used for each what I felt would be semantically correct in a sentence.

If you would rather it be consistent, I suppose going plural would be better.... But for some of those it doesn't make sense (Ex. "Go harvest my mints" implies that there are multiple types of mint, vs one mint plant with multiple leafs to pick.)

**Comment by alexgleason** [comment]
Okay, I created a management command that does this. Here's an example of the output. It skips crops that already exist with the given name. I still need to know what to do about singular vs plural. I'm also leaving this open because I plan to make it automatically happen during the first migration, but I won't do that until the fields for 1.0.0 are finalized.

```
alex@POW-Block:~/Projects/GardenHub$ ./dev.sh manage.py populate_starter_crops
gardenhub_pgdata
Database is listening at port 32777
acorn squash already exists. Skipping.
alfalfa spouts already exists. Skipping.
amaranth already exists. Skipping.
apricots already exists. Skipping.
artichoke already exists. Skipping.
asparagus already exists. Skipping.
avocado already exists. Skipping.
bananas already exists. Skipping.
barley already exists. Skipping.
basil already exists. Skipping.
Created bay leaf.
Created bean sprouts.
Created beets.
Created blackberries.
Created blueberries.
Created bok choy.
broccoli already exists. Skipping.
Created brussels sprouts.
Created butternut squash.
Created cabbage.
Created caper.
carrots already exists. Skipping.
cauliflower already exists. Skipping.
Created cayenne peppers.
Created celery.
Created chamomile.
Created cherries.
Created cherry tomatoes.
Created chestnuts.
Created chickpeas.
Created chives.
cilantro already exists. Skipping.
Created collard greens.
Created coriander.
Created corn.
Created cranberries.
cucumber already exists. Skipping.
Created daisy.
Created daylily.
Created dill.
Created eggplant.
Created elderberries.
Created fennel.
Created garlic.
Created ginger.
Created gourds.
Created grapefruit.
Created green apples.
green beans already exists. Skipping.
Created green grapes.
green pepper already exists. Skipping.
Created habanero pepper.
Created honeydew.
hops already exists. Skipping.
Created hot peppers.
Created huckleberries.
Created jalapeno peppers.
Created kale.
Created kiwi.
Created kohlrabi.
Created leeks.
Created lentils.
lettuce already exists. Skipping.
Created luffa.
Created marigold.
Created mustard.
Created okra.
Created olives.
Created onion.
Created oranges.
Created oregano.
Created oyster mushrooms.
Created parsley.
Created peach.
Created peanuts.
Created pears.
Created peas.
Created peppermint.
Created pineapple.
Created portobello mushrooms.
Created potatoes.
pumpkin already exists. Skipping.
Created purple grapes.
Created quinoa.
Created radish.
Created raspberries.
Created red apple.
red lettuce already exists. Skipping.
Created red onions.
red pepper already exists. Skipping.
Created rosemary.
Created sage.
Created scallions.
Created sesame.
Created shallots.
Created shitake mushrooms.
snap peas already exists. Skipping.
Created snow peas.
Created soy beans.
Created spinach.
Created starfruit.
Created strawberries.
Created sunflower.
Created sweet corn.
Created sweet potatoes.
Created thai basil.
Created thyme.
Created tomatillo.
Created tomatoes.
Created truffle.
Created tulips.
Created walnuts.
Created water melon.
Created white onion.
Created yams.
yellow onion already exists. Skipping.
Created yellow peppers.
zucchini already exists. Skipping.
Finished creating crops!
```

**Comment by marykatefain** [comment]
When we push the new db up for going live just be sure that it actually does re-write over test files. I was very picky with the images I used this time, both in the look and the license. During testing I have been less meticulous so those images should be replaced before going live. 

See note about singular vs/plural above ^

**Comment by alexgleason** [comment]
> be sure that it actually does re-write over test files.

I'm going to nuke the entire `media` directory and database on the first launch. The management command I built requires [crop images to be shipped with the repo](https://github.com/HarvestHub/GardenHub/tree/master/crops). When the management command runs, it loops through each crop in the [`crops_list`](https://github.com/HarvestHub/GardenHub/blob/master/gardenhub/management/commands/_crops_list.py), creates it, then copies the relevant image into the field.

In other words, during the first launch we'll start with an empty database and a list of crops. That's it.

> See note about singular vs/plural above ^

Okay, I had considered that some crops sound different in different contexts. I keep wondering if we need to have multiple fields for different types of crop names. Like: `{title: "tomato", title_plural: "tomatoes"}`. But I haven't fully thought it out. Why do some crops require the singular name vs the plural one even in the same context?

**Comment by marykatefain** [comment]
Sounds good about the db. ^

> Why do some crops require the singular name vs the plural one even in the same context?

English is a weird language. I'm not sure maintaining 2 different versions of the word is worth it, esp because context won't necessarily tell us which version to use. I thought about this a bunch when I was creating the files and this is the conclusion I landed on.

**Comment by alexgleason** [comment]
Alright, I'll trust your judgement on it. Still leaving this open until I add it into the migrations.

I did change "water melon" to "watermelon" if that's alright with you :stuck_out_tongue_winking_eye:

**Comment by marykatefain** [comment]
Haha okay. I'm open to arguments on any individual crop being one way or the other, too. But I do think this is the best way in general to do it. Some could definitely go either way like "onion/s" or "pepper/s".

**Comment by alexgleason** [comment]
Giving up on putting this in a migration. Turns out migrations are run prior to apps being fully loaded, so I can't run a management command from inside one. Not the end of the world if a sysadmin has to run an extra command to set up the initial list of crops.
